packages/frontend-plugin-api/src/types.ts

```diff
@@ -14,16 +14,18 @@
  * limitations under the License.
  */
 
-import { ReactNode } from 'react';
+import { PropsWithChildren } from 'react';
 import { FrontendPlugin } from './wiring';
 
 /** @public */
 export type ProgressProps = {};
 
 /** @public */
-export type NotFoundErrorPageProps = {
-  children?: ReactNode;
-};
+export type NotFoundErrorPageProps = PropsWithChildren<{
+  status?: string;
```

Comment:
This feels a bit strange to have for a component called `NotFoundErrorPage` as it should really always be 404.

I'm wondering if we should actually call this `ErrorPage` in the new frontend system instead as the swappable component. Kind of happy for this to be a bit weird for the old legacy system as that's going away and don't wanna make any breaking changes there.

/cc @freben @patriko thoughts?

Reply:
So in summary - in the new frontend system, having a dedicated NotFoundErrorPage (without any complex props) feels reasonable. And adding a separate ErrorPage with more props later on would also be OK.